c_api/blockingconcurrentqueue.cpp

```diff
@@ -0,0 +1,49 @@
+#include "concurrentqueue.h"
+#include "../blockingconcurrentqueue.h"
+
+typedef moodycamel::BlockingConcurrentQueue<void*> BCQType, *BCQPtr;
+
+extern "C" {
+	
+int moodycamel_bcq_create(BCQHandle* handle)
+{
+	if (handle == nullptr) {
+		return MOODYCAMEL_BAD_HANDLE;
+	}
+	BCQPtr retval = new BCQType;
+	*handle = retval;
+	return MOODYCAMEL_OK;
+}
+
+int moodycamel_bcq_destroy(BCQHandle handle)
+{
+	BCQPtr retval = reinterpret_cast<BCQPtr>(handle);
+	delete retval;
+	return MOODYCAMEL_OK;
+}
+
+int moodycamel_bcq_enqueue(BCQHandle handle, CQValue value)
+{
+	if (handle == nullptr) {
+		return MOODYCAMEL_BAD_HANDLE;
+	}
+	if (value == nullptr) {
```

Comment:
It could be useful to allow a null pointer to be enqueued.

Reply:
only made sense when passing in a pointer to a pointer.